conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1379 @@
+    int[]|[byte...] _ = [1, 2, 3]; // @error ambiguous type '(int[]|[byte...])'
+
+    int[2]|[int, int, int] _ = [1]; // @error ambiguous type '(int[2]|[int,int,int])'
+}
+
+Test-Case: error
+Description: Test the compile-time error when the applicable contextually expected type does not leave
+             a single list type descriptor, with non-list types as union members.
+Labels: boolean, list-constructor-expr, string, union-type
+
+function errorFunction() {
+    int|string|boolean _ = [1, 2, 3]; // @error incompatible types: expected '(int|string|boolean)', found '[int,int,int]'
+}
+
+Test-Case: error
+Description: Test the compile-time error when the applicable contextually expected type does not leave a single list type descriptor,
+             although it is possible to narrow down the type to a single array type considering the list constructor members.
+Fail-Issue: ballerina-platform/ballerina-lang#33056
+Labels: array-type, boolean, list-constructor-expr, string, union-type
+
+function errorFunction() {
+    string[]|int[]|boolean[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    string[]|int[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+}
+
+Test-Case: error
+Description: Test the compile-time error when the applicable contextually expected type does not leave a single list type descriptor,
+             although it is possible to narrow down the type to a single tuple type considering the list constructor members.
+Fail-Issue: ballerina-platform/ballerina-lang#33056
+Labels: boolean, list-constructor-expr, string, tuple-type, union-type
+
+function errorFunction() {
+    [int, int, int]|[string...] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    [int, int, int]|[string...]|int|boolean _ = [1, 2, 3]; // @error inherent type cannot be a union type
+}
+
+Test-Case: error
+Description: Test the compile-time error when the applicable contextually expected type does not leave a single list type descriptor,
+             although it is possible to narrow down the type to a single list type considering the list constructor members.
+Fail-Issue: ballerina-platform/ballerina-lang#33056
+Labels: array-type, boolean, list-constructor-expr, tuple-type, union-type
+
+function errorFunction() {
+    [int, int, int]|boolean[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    int[]|[boolean] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+}
+
+Test-Case: output
+Description: Test when there is no contextually expected type, then the inherent type is a a tuple-type-descriptor
+             with a member-type-descriptor for each expression in the expr-list,
+             where the type of each member-type-descriptor will be the broad type of the corresponding expression
+             in the expr-list.
+Labels: boolean-literal, DecimalNumber, is-expr, list-constructor-expr, singleton-type, string, tuple-type, var
+
+function init() {
+    var a1 = [1, 2, 3];
```

Comment:
Are these the only inherent type tests we have with no contextually-expected type?

Can we add tests like

```ballerina
var a = [1, "foo", "bar", 3];
a[1] = 2; // error
```

to make sure it's `[int, string, string, int]` and not `(int|string)[4]`.